packages/hdwallet-provider/src/index.ts

```diff
@@ -37,13 +57,15 @@ class HDWalletProvider {
   public engine: ProviderEngine;
 
   constructor(
-    mnemonic: string | string[],
+    mnemonic: string | string[] | MnemonicArgs | PrivateKeyArgs,
```

Comment:
There might be a cleaner way to do this with [overloads](https://www.typescriptlang.org/docs/handbook/functions.html#overloads);

(I'm not a TS expert though, please leave a comment if you have an opinion)